proc/split/split.go

```diff
@@ -14,113 +12,25 @@ import (
 // This scheme implements flow control since the SplitProc prevents any of
 // the downstream from running ahead, esentially running the parallel paths
 // at the rate of the slowest consumer.
-type Splitter struct {
-	parent   proc.Interface
-	once     sync.Once
-	n        int
-	requests chan (chan<- proc.Result)
-}
-
-func New(parent proc.Interface) *Splitter {
-	return &Splitter{
-		parent:   parent,
-		requests: make(chan (chan<- proc.Result)),
-	}
-}
-
-func (s *Splitter) Add() chan (chan<- proc.Result) {
-	s.n++
-	return s.requests
-}
-
-func (s *Splitter) gather(strip []chan<- proc.Result) []chan<- proc.Result {
-	flight := strip[:0]
-	for len(flight) < s.n {
-		ch := <-s.requests
-		if ch == nil {
-			s.n--
-		} else {
-			flight = append(flight, ch)
-		}
-	}
-	return flight
-}
-
-func (s *Splitter) run() {
-	// This loop is started by the first downstream SplitChannel as
-	// long as there are active downstream consumers.
-	// If the downstream proc isn't ready, it's request won't have arrived
-	// and this thread will block in the gather loop waiting for all requests.
-	// Once all the requests are available (or null requests are received
-	// indicating the downstream proc is done), then data is pulled from
-	// the upstream path and a reference-counted batch is transmitted to
-	// each requesting entity.
-	strip := make([]chan<- proc.Result, 0, s.n)
-	for {
-		flight := s.gather(strip)
-		if s.n == 0 {
-			break
-		}
-		batch, err := s.parent.Pull()
-		send(flight, proc.Result{batch, err})
-		if batch != nil {
-			batch.Unref()
-		}
-	}
-	s.parent.Done()
-}
-
-func send(flight []chan<- proc.Result, result proc.Result) {
-	for _, ch := range flight {
-		if result.Batch != nil {
-			result.Batch.Ref()
-		}
-		ch <- result
-	}
-}
+type Splitter []proc.Interface
```

Comment:
This thing's only used in this package.
```suggestion
type splitter []proc.Interface
```